### PR TITLE
doc: Rework build-unix.md

### PR DESCRIPTION
**Pull request by maflcko**
The doc has many issues:

* The fist section contains outdated non-existing and confusing configure flags like `--enable-cxx` and `--disable-shared`, as well as edge-case expert options such as `BDB_PREFIX`. Fix that by removing the section and adding notes elsewhere, if applicable.
* There are links to the depends system before instructions on how to simply build from system packages. Fix that by moving that later.
* Also, remove sections that are duplicate with other depends READMEs.

### CODE REVIEW

File: doc/build-unix.md [hunk 1]
```diff
@@ -166,6 +156,12 @@ libqrencode (optional) can be installed with:
 Once these are installed, they will be found by configure and a bitcoin-qt executable will be
 built by default.
 
+## Dependencies
+
+See [dependencies.md](dependencies.md) for a complete overview, and
+[depends](/depends/README.md) on how to compile them yourself, if you wish to
+not use the packages of your Linux distribution.
+
 Notes
```
Comment (fanquake) [inline doc/build-unix.md:165]: While here, I think you could also remove the `Notes`, as well as the `miniupnpc` and `libnatpmp` sections below. I don't think `Notes` really adds any value, and the  `miniupnpc` and `libnatpmp` seem irrelevant here (we also don't have a similar section for all other dependencies (nor do I think we should add one), and listing configure defaults in markdown is mostly pointless, and just duplicates the (always up to date) output of `./configure --help`).

We could also remove the `Security` section further down, I don't think this document is the right place for generic, binary hardening information.

**Review by fanquake — COMMENTED** [review]
Concept ACK - build-unix probably a good candidate for being split up further at some point.

File: doc/build-unix.md (hunk 1 above)
Comment (fanquake) [reply doc/build-unix.md]: Didn't drop `Security` ?

Comment (fanquake) [reply doc/build-unix.md]: Ok. Done in #27688.

**Review by fanquake — APPROVED** [review]
ACK fa29651c3ff3e13a42d6505b14971265562f088a

**Review by sedited — APPROVED** [review]
ACK fa29651c3ff3e13a42d6505b14971265562f088a

**Pull request merged**